hw/xbox/xid-sdl.c

```diff
@@ -194,36 +199,31 @@ static const XIDDesc desc_xid_xbox_gamepad = {
 
 #define BUTTON_MASK(button) (1 << ((button) - GAMEPAD_DPAD_UP))
 
-static void update_output(USBXIDState *s) {
-#ifdef FORCE_FEEDBACK
-    if (s->sdl_haptic == NULL) { return; }
+static int haptic_effect_init(SDL_HapticEffect *effect, HapticEmulationMode mode, uint16_t large_magnitude, uint16_t small_magnitude) {
+    switch (mode) {
+    case EMU_HAPTIC_LEFT_RIGHT:
+        effect->leftright = (SDL_HapticLeftRight){
+            .type = SDL_HAPTIC_LEFTRIGHT,
+            .length = SDL_HAPTIC_INFINITY,
+            .large_magnitude = large_magnitude >> 1,
```

Comment:
Probably worth mentioning somewhere in a comment that the range Xbox magnitude is in range [0-`0xffff`] (according to your research) while SDL apparently has a [max value of `0x7fff`](https://wiki.libsdl.org/SDL_HapticEffect#Remarks), hence the shift.